src/hotspot/share/cds/filemap.cpp

```diff
@@ -1602,14 +1602,17 @@ char* FileMapInfo::write_bitmap_region(const CHeapBitMap* ptrmap, ArchiveHeapInf
     size_in_bytes += heap_info->ptrmap()->size_in_bytes();
   }
 
-  // The bitmap region contains up to 3 parts:
-  // ptrmap:              metaspace pointers inside the ro/rw regions
+  // The bitmap region contains up to 4 parts:
+  // rw_ptrmap:              metaspace pointers inside the read-write region
+  // ro_ptrmap:              metaspace pointers inside the read-only region
   // heap_info->oopmap(): Java oop pointers in the heap region
   // heap_info->ptrmap(): metaspace pointers in the heap region
   char* buffer = NEW_C_HEAP_ARRAY(char, size_in_bytes, mtClassShared);
   size_t written = 0;
-  written = write_bitmap(ptrmap, buffer, written);
-  header()->set_ptrmap_size_in_bits(ptrmap->size());
+  written = write_bitmap(rw_ptrmap, buffer, written);
+  header()->set_rw_ptrmap_size_in_bits(rw_ptrmap->size());
+  written = write_bitmap(ro_ptrmap, buffer, written);
+  header()->set_ro_ptrmap_size_in_bits(ro_ptrmap->size());
```

Comment:
We actually have fields in `CDSFileMapRegion` that record the size and starting position of a bitmap inside the `bm` region: (the comments for `_ptrmap_offset` needs to be updated).

```
typedef struct CDSFileMapRegion {
...
  size_t  _ptrmap_offset;     // Bitmap for relocating native pointer fields in archived heap objects.
                              // (The base address is the bottom of the BM region).
  size_t  _ptrmap_size_in_bits;
```

I think these can be reused to store the information for the ro_ptrmap and rw_ptrmap.

```
  region_at(MetaspaceShared::rw)->init_oopmap(0, rw_ptrmap->size());
  written = write_bitmap(rw_ptrmap, buffer, written);

  region_at(MetaspaceShared::ro)->init_oopmap(written, ro_ptrmap->size());
  written = write_bitmap(ro_ptrmap, buffer, written);
```